eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -392,42 +464,166 @@ function Invoke-GenerateAndBuildSDK () {
         $projectFolder = $newpackageoutputJson.projectFolder
         $path = $newpackageoutputJson.path
         $service = $newpackageoutputJson.service
-        $packageName = $newpackageoutputJson.packageName
+        # $packageName = $newpackageoutputJson.packageName
         Write-Host "projectFolder:$projectFolder"
 
-        # Generate Code
-        Invoke-Generate -sdkfolder $projectFolder
-        if ( !$?) {
-            Write-Error "Failed to generate sdk. exit code: $?"
-            exit 1
-        }
+        GeneratePackage -projectFolder $projectFolder -sdkRootPath $sdkRootPath -path $path -downloadUrlPrefix $downloadUrlPrefix -generatedSDKPackages $generatedSDKPackages
+    }
+}
 
+function GeneratePackage()
+{
+    param(
+        [string]$projectFolder,
+        [string]$sdkRootPath,
+        [string]$path,
+        [string]$downloadUrlPrefix="",
+        [object]$generatedSDKPackages
+    )
+
+    $packageName = Split-Path $projectFolder -Leaf
+    $projectFolder = $projectFolder -replace "\\", "/"
+    $projectFolder -match "sdk/(?<service>.*)/"
+    $service = $matches["service"]
+    Write-Host "Generating code for " $packageName
+    $artifacts = @()
+    $apiViewArtifact = ""
+    $hasBreakingChange = $null
+    $content = $null
+    $result = "succeeded"
+
+    # Generate Code
+    Write-Host "Start to generate sdk $projectFolder"
+    Invoke-Generate -sdkfolder $projectFolder
+    if ( !$?) {
+        Write-Error "Failed to generate sdk. exit code: $?"
+        $result = "failed"
+    } else {
         # Build
+        Write-Host "Start to build sdk: $projectFolder"
         Invoke-Build -sdkfolder $projectFolder
         if ( !$? ) {
             Write-Error "Failed to build sdk. exit code: $?"
-            exit 1
+            $result = "failed"
+        } else {
+            # pack
+            Write-Host "Start to pack sdk"
+            Invoke-Pack -sdkfolder $projectFolder
+            if ( !$? ) {
+                Write-Error "Failed to packe sdk. exit code: $?"
+            }
+            # Generate APIs
+            Write-Host "Start to export api for $service"
+            pwsh $sdkRootPath/eng/scripts/Export-API.ps1 $service
```

Comment:
Why are you starting a new `pwsh` when you could just run the script? It's much more expensive. And if you set the `$ErrorActionPreference = 'Stop'` it will terminate the whole session automatically without having to check `$?`.

Reply:
update to use `&` to call powershell script